src/ui/shared/requests/useWalletsMetaByChunks.ts

```diff
@@ -1,22 +1,39 @@
 import { useQuery } from '@tanstack/react-query';
 import { ZerionAPI } from 'src/modules/zerion-api/zerion-api.client';
+import { normalizeAddress } from 'src/shared/normalizeAddress';
+import { queryClient } from './queryClient';
 
 export function useWalletsMetaByChunks({
   addresses,
   enabled = true,
   suspense = true,
   useErrorBoundary = true,
+  staleTime,
 }: {
   addresses: string[];
   enabled?: boolean;
   suspense?: boolean;
   useErrorBoundary?: boolean;
+  staleTime?: number;
 }) {
   return useQuery({
     enabled: enabled && addresses.length > 0,
     queryKey: ['ZerionAPI.getWalletsMetaByChunks', addresses],
-    queryFn: () => ZerionAPI.getWalletsMetaByChunks(addresses),
+    queryFn: async () => {
+      const result = await ZerionAPI.getWalletsMetaByChunks(addresses);
+      if (result.length > 1) {
+        result.forEach((walletMeta) => {
+          const normalizedAddress = normalizeAddress(walletMeta.address);
+          queryClient.setQueryData(
+            ['ZerionAPI.getWalletsMetaByChunks', [normalizedAddress]],
```

Comment:
Why is "by chunks" if it's a single address?
I think it's more appropriate to set this as `['walletGetMeta', normalizedAddress]`

Reply:
Let's research `initialData` approach? https://tkdodo.eu/blog/seeding-the-query-cache#seeding-details-from-lists

Reply:
Because the logic is to avoid single address fetches if we've already fetched a group
I will rename the fetcher